coderd/wsbuilder/wsbuilder.go

```diff
@@ -632,19 +637,33 @@ func (b *Builder) getParameters() (names, values []string, err error) {
 		//  conditional parameter existence, the static frame of reference
 		//  is not sufficient. So assume the user is correct, or pull in the
 		//  dynamic param code to find the actual parameters.
+		latestValues := make(map[string]string)
+		for _, latest := range b.richParameterValues {
+			latestValues[latest.Name] = latest.Value
+		}
+
+		// Merge the inputs with values from the previous build.
+		for _, last := range lastBuildParameterValues {
+			// TODO: Ideally we use the resolver here and look at parameter
+			//   fields such as 'ephemeral'. This requires loading the terraform
+			//   files. For now, just send the previous inputs as is.
+			if _, exists := latestValues[last.Name]; exists {
+				continue
+			}
+			names = append(names, last.Name)
+			values = append(values, last.Value)
+		}
+
```

Comment:
I am skipping validation right now for dynamic params experimental. I have an issue to address this again later.

https://github.com/coder/coder/issues/17942